### PR TITLE
Add test for AddEventListenerOptions passive member

### PR DESCRIPTION
**Pull request by RByers**
Also includes a minor style cleanup to the related capture option test.

Includes logic from blink tests for this feature (updated for WPT style): [passive_dispatch.html](https://chromium.googlesource.com/chromium/src/+/b4cc79f5db2ea11ff1c865ce43bca1b035c6ef97/third_party/WebKit/LayoutTests/fast/events/eventlisteneroptions/passive_dispatch.html), [passive_inequality.html](https://chromium.googlesource.com/chromium/src/+/b4cc79f5db2ea11ff1c865ce43bca1b035c6ef97/third_party/WebKit/LayoutTests/fast/events/eventlisteneroptions/passive_inequality.html) and [passive_query.html](https://chromium.googlesource.com/chromium/src/+/b4cc79f5db2ea11ff1c865ce43bca1b035c6ef97/third_party/WebKit/LayoutTests/fast/events/eventlisteneroptions/passive_query.html).

### CODE REVIEW

**Comment by RByers** [comment]
@dtapuska PTAL

File: dom/events/AddEventListenerOptions-passive.html [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+function testPassiveValue(optionsValue, expectedDefaultPrevented) {
+  var defaultPrevented = undefined;
+  var handler = function handler(e) {
+    assert_false(e.defaultPrevented, "Event prematurely marked defaultPrevented");
+    e.preventDefault();
+    defaultPrevented = e.defaultPrevented;
+  }
+  document.addEventListener('test', handler, optionsValue);
+  var uncanceled = document.body.dispatchEvent(new Event('test', {bubbles: true, cancelable: true}));
+
```
Comment (dtapuska) [inline dom/events/AddEventListenerOptions-passive.html]: Should you add an assert that defaultPrevented !== undefined? to ensure that the event was actually dispatched?

File: dom/events/AddEventListenerOptions-passive.html [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+test(function() {
+  // Sanity check options that should be treated as distinct handlers
+  testOptionEquivalence({capture:true}, {capture:false, passive:false}, false);
+  testOptionEquivalence({capture:true}, {passive:true}, false);
+
+  // Option values that should be treated as equivalent
+  testOptionEquivalence({}, {passive:false}, true);
+  testOptionEquivalence({passive:true}, {passive:false}, true);
```
Comment (dtapuska) [inline dom/events/AddEventListenerOptions-passive.html]: {capture: true, passive: false}, {capture: true, passive: true} ?

**Comment by dtapuska** [comment]
lgtm % one question

File: dom/events/AddEventListenerOptions-passive.html (hunk 2 above)
Comment (RByers) [inline dom/events/AddEventListenerOptions-passive.html]: Done, thanks.

**Comment by dtapuska** [comment]
lgtm

**Pull request merged**